include/SourceLocation.h

```diff
@@ -57,6 +58,9 @@ class SourceLocation {
     /// Note that there is no error checking to ensure that the location
     /// still points to a valid place in the source.
     SourceLocation operator+(int delta) const {
+        if (bufferID.getValue() == 6) {
```

Comment:
Oops. I was using this for debugging since this bug somehow caused the location to point to an invalid location in the source. I'll remove it.